### PR TITLE
[Flow decomposition][Reformat] Observer and NetworkUtil

### PR DESCRIPTION
**Pull request by OpenSuze**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

The goal is to remove terminal setP and setQ in losses compensation PR #147.
I used this PR to reformat observers in flow decompostion computer and the NetworkUtils functions.

**What kind of change does this PR introduce?**

Reformating

**What is the current behavior?**

No change

**What is the new behavior (if this is a feature change)?**
No change

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

**Other information**:

### CODE REVIEW

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java
Comment (caioluke) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java:68]: Is the TODO from `.filter(NetworkUtil::isInMainSynchronousComponent)` still valid? It is from 2022

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java
Comment (caioluke) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java:130]:
```suggestion
            .filter(PhaseTapChangerHolder::hasPhaseTapChanger)
```

And remove the `isPst` function

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java
Comment (caioluke) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java:129]: What about ThreeWindingsTransformers? Shouldn't we add them too?
If not, shouldn't we warn the user about that?

**Review by caioluke — COMMENTED** [review]
Thank you for the refactoring and the utils tests. 😄

I think observers are much better placed now!

Question remains about the 2 TODO's and whether or not to include three windings transfomers in `getPstIdList`.

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java
Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java]: Yes, I added some tests to highlight this. I will also write documentation

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java
Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java:69]: Yes, I added some tests to highlight this. I will also write documentation

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java
Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java:150]: Do you have a methodology to decompose their flow ? ThreeWindingsTransformers have a PhaseTapChanger per leg...

Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java:150]: But, we could write some documentation about them

**Comment by OpenSuze** [comment]
I added a lot of test with some "golden master" tests

**Comment by OpenSuze** [comment]
- [ ] Check new TODO
- [ ] Write documentation injection != BusbarSection ShuntCompensator StaticVarCompensator
- [ ] Write documentation main synchronous component
- [ ] Write documentation ThreeWindingsTransformers
- [ ] merge with losses_compensation_improvement branch

**Comment by OpenSuze** [comment]
Thank you for the review, I made some changes. Ready for review

**Review by caioluke — APPROVED** [review]
Wow, thanks a lot for the work you put into this PR :)

I have only some minor comments

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java
Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java:152]: Thank you ! I did not know that :)

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/TestUtils.java
Comment (OpenSuze) [reply flow-decomposition/src/test/java/com/powsybl/flow_decomposition/TestUtils.java:60]: Created https://github.com/powsybl/powsybl-entsoe/issues/150

**Comment by OpenSuze** [comment]
Waiting for a final green light to merge this PR

**Comment by caioluke** [comment]
@OpenSuze it's all good for me!

**Pull request merged**
